xontrib/jedi.py

```diff
@@ -138,7 +138,6 @@ def create_completion(comp):
 xonsh.completers.base.complete_python = complete_jedi
 
 # Jedi ignores leading '@(' and friends
-completer remove python_mode
-
-completer add jedi_python complete_jedi '<python'
-completer remove python
+_aliases._remove_completer(["python_mode"])
+_aliases._add_one_completer("jedi_python", complete_jedi, "<python")
+_aliases._remove_completer(["python"])
```

Comment:
Yeah, I agree. I don't love this.  There probably should be a public API. Maybe that would be best in another PR?

Reply:
Ok. I will create a PR for that later. I will update the tests for this